rootston/keyboard.c

```diff
@@ -87,19 +87,20 @@ static const char *exec_prefix = "exec ";
 
 static void keyboard_binding_execute(struct roots_keyboard *keyboard,
 		const char *command) {
-	struct roots_server *server = keyboard->input->server;
+	struct roots_seat *seat = keyboard->seat;
 	if (strcmp(command, "exit") == 0) {
-		wl_display_terminate(server->wl_display);
+		wl_display_terminate(keyboard->input->server->wl_display);
 	} else if (strcmp(command, "close") == 0) {
-		if (server->desktop->views->length > 0) {
-			struct roots_view *view =
-				server->desktop->views->items[server->desktop->views->length-1];
-			view_close(view);
+		if (!wl_list_empty(&seat->views)) {
```

Comment:
This should act on the focused view. If no view is focused, it should do nothing.

Reply:
You're right. Maybe it makes sense to add `roots_seat_focus_next_view` for the other command?

Reply:
I think that makes sense.